ui/pages/RewardsDashboard.tsx

```diff
@@ -0,0 +1,230 @@
+                  `Claim ${ dailyRewardValue } Merits` :
+                  `Next claim in ${ timeLeft }`
+                }
+              </Button>
+            ) }
+          >
+            <RewardsDashboardCardValue
+              label="Total balance"
+              value={ balancesQuery.data?.total || 0 }
+              isLoading={ balancesQuery.isPending }
+              withIcon
+              hint={ (
+                <>
+                  Total number of merits earned from all activities.{ ' ' }
+                  <LinkExternal href="https://docs.blockscout.com/using-blockscout/my-account/merits">
+                    More info on merits
+                  </LinkExternal>
+                </>
+              ) }
+            />
+          </RewardsDashboardCard>
+          <RewardsDashboardCard
+            title="Referrals"
+            description="Total number of users who have joined the program using your code or referral link."
+            direction="column-reverse"
+          >
+            <RewardsDashboardCardValue
+              label="Referrals"
+              value={ `${ numberOfReferrals } user${ numberOfReferrals === 1 ? '' : 's' }` }
+              isLoading={ referralsQuery.isPending }
+              hint="The number of referrals who registered with your code/link."
+            />
+          </RewardsDashboardCard>
+          <RewardsDashboardCard
+            title="Steaks"
+            description={ `Current number of consecutive days you${ apos }ve claimed your daily Merits.` }
+            direction="column-reverse"
+            availableSoon
+            blurFilter
+          >
+            <RewardsDashboardCardValue label="Steaks" value="5 days"/>
+          </RewardsDashboardCard>
+        </Flex>
+        <RewardsDashboardCard
+          title="Referral program"
+          description={ (
+            <>
+              Refer friends and boost your merits! You receive a{ ' ' }
+              <Skeleton as="span" isLoaded={ !rewardsConfigQuery.isPending }>
+                { Number(rewardsConfigQuery.data?.rewards.referral_share || 0) * 100 }%
+              </Skeleton>
+              { ' ' }bonus on all merits earned by your referrals.
+            </>
+          ) }
+          direction="row"
+        >
+          <Flex
+            flex={ 1 }
+            gap={{ base: 2, md: 6 }}
+            px={{ base: 4, md: 6 }}
+            py={{ base: 4, md: 0 }}
+            flexDirection={{ base: 'column', md: 'row' }}
+          >
+            <CopyField
+              label="Referral link"
+              value={ `https://eth.blockscout.com?ref=${ referralsQuery.data?.code }` }
+              isLoading={ referralsQuery.isPending }
+              flex={ 2 }
+            />
+            <CopyField
+              label="Referral code"
+              value={ referralsQuery.data?.code || '' }
+              isLoading={ referralsQuery.isPending }
+              flex={ 1 }
+            />
+          </Flex>
```

Comment:
If there is no referal code, should we still render this section?

Reply:
The referral code should always be there. If it is not there, it is a problem on the service's side